src/main/java/com/google/api/codegen/discogapic/DiscoGapicProviderFactory.java

```diff
@@ -18,12 +18,20 @@
 import com.google.api.codegen.config.PackageMetadataConfig;
 import com.google.api.codegen.discovery.Document;
 import com.google.api.codegen.gapic.GapicGeneratorConfig;
+import com.google.api.codegen.gapic.GapicProvider;
 import java.util.List;
 
+/** A factory for DiscoGapicProviders which perform code generation. */
 public interface DiscoGapicProviderFactory {
-  List<DiscoGapicProvider> create(
+  /**
+   * Create the provider from the given model, configs and the output path.
+   *
+   * <p>The outputPath is used for copying static files from resources into the output directory.
+   */
+  List<GapicProvider<? extends Object>> create(
```

Comment:
Out of curiosity - was there something that forced us to make the element type here more generic?

Reply:
The original GapicProviders in that returned list were DiscoGapicProviders. However, then I wanted to add a StaticGapicProvider (line 131ish) which was just a GapicProvider , so I just generalized the List inner type to GapicProviders.